src/Support/Helpers/ModelHelper.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tpetry\PostgresqlEnhanced\Support\Helpers;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\SoftDeletes;
+
+class ModelHelper
+{
+    public function getCreatedAtColumn(Model $model): ?string
+    {
+        return match ($model->usesTimestamps()) {
+            true => $model->getCreatedAtColumn(),
+            false => null,
+        };
+    }
+
+    public function getDeletedAtColumn(Model $model): ?string
+    {
+        if (!\in_array(SoftDeletes::class, trait_uses_recursive($model::class)) || !method_exists($model, 'isForceDeleting') || !method_exists($model, 'getDeletedAtColumn')) {
+            return null;
+        }
+        if ($model->isForceDeleting()) {
+            return null;
+        }
```

Comment:
It does not makes sense here to check `isForceDeleting` as the property won't be true at any moment (set to true only when `Model::forceDelete()` is called)